src/components/footers/Footer.tsx

```diff
@@ -0,0 +1,62 @@
+import { FooterLinks } from '../molecules/footers/FooterLinks'
+import { navLinks, roles, genres } from '../../common/constants/enums'
+import { BottomIcons } from '../molecules/footers/BottomIcons'
+
+export function Footer() {
+  const generateColumnData = (
+    enumValues: string[],
+    title: string,
+    prefix: string = ''
+  ) => {
+    const items = enumValues.map(value => ({
+      label: value.replace('_', ' '),
+      link: `/${prefix}${value.toLowerCase().replace('_', '-')}`,
+    }))
+    return { title, items }
+  }
```

Comment:
❓ Any reason this function can't be defined outside of the component?

Reply:
Where would you prefer to see it?

Reply:
When you have a function that gets all of its variables as input (like this), I prefer to place it just outside the component function. IF it's going to be used by multiple components, then place it in some common utils file.

```typescript
function MyComponent({someNumber}) {
  function functionThatCannotBeDefinedOutsideOfComponent() {
    return 2 + someNumber;
  }
}

function functionThatCanBeDefinedOutsideOfComponent(number) {
  return 2 + number
}
```

Reply:
Makes perfect sense!
Thanks